### ISSUE TITLE
Script/Env files for local setup

### ISSUE THREAD

**Issue by mathiaszinnen** [jvalue/ods#230]
To enable easier and faster local testing outside of the docker environment, adding a script to set the needed environment variables might be helpful. Another approach is using local .env files. We should find a common solution for all services and implement that. 
Do you have any thoughts on that @georg-schwarz , @andreas-bauer @lunedis @jsone-studios @jenswaechtler

**Comment by sonallux** [comment]
> adding a script to set the needed environment variables

This would be the simplest solution, but having a shell script would not work on windows. So cross-platform compatibility is difficult with this approach, as different scripts need to be maintained.

> Another approach is using local .env files.

For `.env` file loading, we can use the [dotenv](https://www.npmjs.com/package/dotenv) package. Then a `.env` and add a new command to each `package.json`: `"start-local": "node -r dotenv/config --enable-source-maps dist/index.js"`

---
I would prefer option 2 with the `.env` files. To get a local setup working, we also must make sure all dependent services are accessible from outside the docker-compose environment. For example, all the Postgres databases are currently not exposed to the outside.

**Comment by georg-schwarz** [comment]
Is it really an issue? I kind of never start up the services locally but only in Docker.
 
Or do you mean for debugging? Then I'd also strongly tend to a `.env.local` file. Though it is something that has to be kept synced and so on...

**Comment by mathiaszinnen** [comment]
It is for both, local development and debugging. I think having to start the services in the docker environment including multi stage rebuild etc. slows the development down considerably. That was the reason I added a script in the first place

**Comment by sonallux** [comment]
When optimizing the docker build we should also consider removing the linting and testing steps from the Dockerfile for local builds as they also take quite some time. Maybe we can add something like a flag to disable the linting and testing.

**Comment by georg-schwarz** [comment]
I think  @andreas-bauer is putting a little research into that. IMHO the node projects are not the big issue here... Just Gradle loads forever...

So my fear in this discussion that we will have another few config files that are not kept in sync when changes are made. Somehow we have to overcome that issue.

Here are some further thoughts:
* starting up the integration tests in a docker container should not be the deal-breaker. It is already considerably fast, and there should not be too much need to debug it
* starting up the service itself and let the integration tests run against it is more a common use case I would assume. So I see here the real need to get it running locally with integration tests from a docker container
* this should be doable by changing the URL to the service in the it env variable, and starting all dependencies of the service via `docker-compose up -d service && docker-compose stop service`, then start the integration test against the service as usual.

Opinions?

**Comment by mathiaszinnen** [comment]
I think optimizing the docker build is definitely something that will make development easier/faster. So we should absolutely do that. 
However, I still prefer running the services or the integration tests locally in some cases. Mainly for debugging, but also for instant feedback. This is the reason I created the script that triggered the discussion. I uploaded it because I thought it might be useful for somebody else who wants to run the service locally, too.
As long as I am the only one using it, I would suggest I just use it locally without uploading it so we do not have to maintain it. 
We can reopen the issue if there are others who also want to have it running outside of the docker environment.
